.github/workflows/python-release.yaml

```diff
@@ -43,14 +39,42 @@ jobs:
       - uses: actions/setup-python@v5
         with:
           python-version: ${{ env.PYTHON_VERSION }}
-      - name: Build wheels
+          architecture: x64
+      - name: Build wheels - x86_64
         uses: PyO3/maturin-action@v1
         with:
-          target: ${{ matrix.target }}-apple-darwin
-          args: --release --out dist
+          args: --out dist
           working-directory: bindings/python
           sccache: "true"
           container: "off"
+          maturin-version: ${{ env.MATURIN_VERSION }}
+      - name: Upload wheels
+        uses: actions/upload-artifact@v3
+        with:
+          name: wheels
+          path: bindings/python/dist
+
+  macos-universal:
+    runs-on: macos-14
+    steps:
+      - uses: actions/checkout@v4
+      - uses: extractions/setup-just@v1
+        with:
+          just-version: "1.23.0"
+      - name: install protoc
+        run: just protoc
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+          architecture: x64
+      - name: Build wheels - universal
+        uses: PyO3/maturin-action@v1
+        with:
+          args: --target universal2-apple-darwin --out dist
+          working-directory: bindings/python
+          sccache: "true"
```

Comment:
I think we don't want this. We want caches to be off for release builds (to avoid having something unexpected from a previous build to leak in,) and even so it we're pretty unlikely to be able to use the cache, and it would take some amount of time to upload.